cluster/cluster.go

```diff
@@ -79,20 +79,31 @@ func Start() {
 	log.Info("CLU Start: joined to %d nodes in cluster", n)
 }
 
+type partitionCandidates struct {
+	priority int
+	nodes    []Node
+}
+
 // return the list of nodes to broadcast requests to
-// Only 1 member per partition is returned. This list includes
-// ThisNode if it is capable of handling queries.
+// Only 1 member per partition is returned.
+// The nodes are selected based on priority, using a random selection
+// if there are multi nodes with the highest priority.
 func MembersForQuery() []Node {
 	thisNode := Manager.ThisNode()
 	// If we are running in single mode, just return thisNode
 	if Mode == ModeSingle {
 		return []Node{thisNode}
 	}
 
-	membersMap := make(map[int32][]Node)
+	// store the available nodes for each partition, grouped by
+	// priority
+	membersMap := make(map[int32]*partitionCandidates)
 	if thisNode.IsReady() {
```

Comment:
I realized that though a node can handle requests (forwarding to other nodes) while it is low prio, we use the ready state in hosted-metrics for the k8s readiness probe. 
This has complications for deploying upgrades.  We do rolling upgrades, but if the nodes report as being ready when they cant actually serve data then k8s will push the upgrade out too fast and there will be no instances with all of the data (all running instances will still be processing their backlog)

So in addition to converting the lag from a number of messages behind, i have also added a config option for maxPrio to mark the node as notReady.  By default it is set to 10seconds, so if the node is 10secon or more behind it will be marked as notReady.  For hosted-metrics this will ensure that the nodes remain out of service until they have completed processing the backlog of data (or within 10seconds of processing it)/